### PR TITLE
Modify repositories/rhel-xy.json file before testing nightly compose

### PR DESCRIPTION
**Pull request by atodorov**
- uses a wildcard match for sed in order to replace only lines which
  match and keep the rest intact

This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

File: tools/provision.sh [hunk 1]
```diff
@@ -174,3 +168,10 @@ case "${AUTH_METHOD}" in
         sudo systemctl start osbuild-composer.socket
         ;;
 esac
+
+# Print debugging info about content sources
+sudo composer-cli status show
```
Comment (ondrejbudai) [inline tools/provision.sh:173]: This should be moved under the `AUTH_METHOD_NONE` section, tests meant for the service don't have Weldr enabled at all.

**Comment by atodorov** [comment]
Nightly test job is here:
https://gitlab.com/redhat/services/products/image-builder/ci/osbuild-composer/-/pipelines/613845213

failures are unrelated.

**Review by jrusz — CHANGES_REQUESTED** [review]
> * the schutzbot/prepare-rhel-internal-json.sh script will be executed
>   before mockbuild.sh

How and why? Looking at the `.gitlab-ci.yml` file I can see it's running the `mockbuild.sh` first to create the test rpms and the it's running the `prepare-rhel-internal` stage...

Also this means that currently the `prepare-rhel-internal` is just making sure the correct system repos are set and optinally a custom `-tests` package repo url is used, right?

Otherwise it looks good I like the simple `sed` to replace the source repos during provisining of the SUT.

**Comment by jrusz** [comment]
@atodorov the issue is that this https://github.com/osbuild/osbuild-composer/blob/main/test/cases/regression-composer-works-behind-satellite.sh#L127 doesn't return anything

Latest run on main: https://gitlab.com/redhat/services/products/image-builder/ci/osbuild-composer/-/jobs/2887431645#L3037
Run with this PR:  https://gitlab.com/redhat/services/products/image-builder/ci/osbuild-composer/-/jobs/2890016059#L3183

And the reason seems to be this one:https://github.com/osbuild/osbuild-composer/blob/main/test/cases/regression-composer-works-behind-satellite.sh#L101 
It's expecting the repo name to be all lowercase in nightly run

**Comment by atodorov** [comment]
> > * the schutzbot/prepare-rhel-internal-json.sh script will be executed
> >   before mockbuild.sh
> 
> How and why? Looking at the `.gitlab-ci.yml` file I can see it's running the `mockbuild.sh` first to create the test rpms and the it's running the `prepare-rhel-internal` stage...
> 

^^^ no longerr valid, that was an initial approach. Commit log message updated to match.


> Also this means that currently the `prepare-rhel-internal` is just making sure the correct system repos are set and optinally a custom `-tests` package repo url is used, right?
> 

Yes.

> Otherwise it looks good I like the simple `sed` to replace the source repos during provisining of the SUT.

**Comment by atodorov** [comment]
> And the reason seems to be this one:https://github.com/osbuild/osbuild-composer/blob/main/test/cases/regression-composer-works-behind-satellite.sh#L101

Good catch, thanks. I didn't scroll all the way down apparently.  Fixed in the latest commit.

**Comment by atodorov** [comment]
9.1 test job:
https://gitlab.com/redhat/services/products/image-builder/ci/osbuild-composer/-/pipelines/615634410

8.7 test job:
https://gitlab.com/redhat/services/products/image-builder/ci/osbuild-composer/-/pipelines/615634450

**Review by jrusz — APPROVED** [review]
Alright, nightly pipelines looks good, so LGTM

**Pull request merged**